deployer/src/deployer/upload.py

```diff
@@ -603,6 +626,42 @@ def on_task_complete(task):
                 if delta.days < 30:
                     continue
 
+            # Remember, if `key` is from a "index.html" file it will be represented
+            # something like this: `main/en-us/docs/web/api/documentorshadowroot`
+            # with the prefix and the `/index.html` portion removed.
+            # But every page usually has a `index.json` file, which might look
+            # something like this: `main/en-us/docs/web/api/index.json` or
+            # `main/en-us/docs/web/api/screenshot.png`
+
+            # This if statement protects against possible deleting anything that
+            # isn't a document.
+            if "/docs/" in key:
+                is_archived = False
+                # Trying to avoid having to do another for-loop with key.startswith()
+                # so first look for the low-hanging fruit.
+                if key in archived_files_as_keys:
+                    # This is the easiest and fastest lookup
+                    is_archived = True
+                elif (
+                    re.sub(r"/(index.json|contributors.txt|bcd.json)$", "", key)
```

Comment:
Since `.` matches any character instead of just a "period", this will also match for example`.../index-json`, so should be:
```suggestion
                    re.sub(r"/(index\.json|contributors\.txt|bcd\.json)$", "", key)
```